### PR TITLE
[WIP] dataframe docs and cleanup

### PR DESCRIPTION
**Pull request by cowlicks**

### CODE REVIEW

File: docs/source/frame-creation.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    >>> from dask.dataframe.io import from_array
+    >>> x = np.array([(i, i*10) for i in range(10)],
+                     dtype=[('a', 'i4'), ('b', ('i4')]))
+    >>> dd.from_array(x, chunksize=4)
+
+
+From a CSV file.
+-----------------
+
+beep
```
Comment (mrocklin) [inline docs/source/frame-creation.rst]: I can handle these TODOs.  This would definitely be a useful doc to have.

**Comment by cowlicks** [comment]
@mrocklin is it dask `Frame` or dask `DataFrame`?

**Comment by mrocklin** [comment]
Used to be called frame, have since renamed to DataFrame

**Comment by mrocklin** [comment]
Closed by #239

**Pull request closed without merging**